client_types.go

```diff
@@ -170,3 +171,39 @@ func (m Memory) String() string {
 	return fmt.Sprintf("Total: %d, Available: %d, Used: %d, Free: %d",
 		m.Total, m.Available, m.Used, m.Free)
 }
+
+type ClientType struct {
+	Name string
+}
```

Comment:
The `ClientType` is a just list of protocols. Adding makes code more complex for no good reason. It actually makes the APIs less useful. 

For example:

```go
func (c *Client) listSharedTunnels(ct ClientType) (map[string][]TunnelState, error)
```

is less flexible than

```
func (c *Client) listSharedTunnels(protocol ...Protocol) (map[string][]TunnelState, error) {
```

Reply:
I agree. I think Alana tries to separate IPSec from Sauce Connect (ie "sc" group contains both kgp and h2c). I don't think it's necessary to group like this. I'd say, if the "protocol" parameter is specified, return only instances for this protocol. Otherwise, return them all. We use "ipsec only" quite a lot because ipsec is managed differently. I don't see a use case where we need everything BUT ipsec